### PR TITLE
Add Poll

### LINKED ISSUE #1064: Introduce Poll as a nominal type for uncancelable's param
It's very difficult to look up scaladoc for `F ~> F`. We would be a lot better off with something like this in kernel:

```scala
trait Poll[F[_]] extends (F ~> F)
```

This gives us a place for documentation.

### PR DESCRIPTION
**Pull request by vasilmkd**
Resolves #1064.

### CODE REVIEW

File: kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala [hunk 1]
```diff
@@ -32,14 +32,16 @@ trait Fiber[F[_], E, A] {
     joinAndEmbed(F.canceled *> F.never)
 }
 
+trait Poll[F[_]] extends (F ~> F)
```
Comment (djspiewak) [inline kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala]: Can you move this to its own file? Arguably `Fiber` should be moved as well, and then **concurrent.scala** can be capitalized to **Concurrent.scala**.

**Review by djspiewak — CHANGES_REQUESTED** [review]
Very nice! Minor request but otherwise good to go

File: kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala (hunk 1 above)
Comment (vasilmkd) [reply kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala]: Finally! 😆

File: kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala [hunk 2]
```diff
@@ -21,19 +21,6 @@ import cats.data.{EitherT, Ior, IorT, Kleisli, OptionT, WriterT}
...
-  def joinAndEmbed(onCancel: F[A])(implicit F: Concurrent[F, E]): F[A] =
-    join.flatMap(_.fold(onCancel, F.raiseError(_), fa => fa))
-
-  def joinAndEmbedNever(implicit F: Concurrent[F, E]): F[A] =
-    joinAndEmbed(F.canceled *> F.never)
-}
-
-trait Poll[F[_]] extends (F ~> F)
-
 trait Concurrent[F[_], E] extends MonadError[F, E] {
```
Comment (djspiewak) [inline kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala:24]: Super-minor thing, but would you mind renaming this file to upper-case?

Comment (vasilmkd) [reply kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala:24]: I swear I did. Git failed me. Fixing.

Edit: Nevermind, Mac OS failed me.

Comment (vasilmkd) [reply kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala:24]: `git mv` to the rescue.

Comment (djspiewak) [reply kernel/shared/src/main/scala/cats/effect/kernel/concurrent.scala:24]: Yeah Mac and Windows are terrible for that kind of thing :-/

**Pull request merged**